### PR TITLE
Fyle t 3khv6k: Critical policy violation popup

### PR DESCRIPTION
**Pull request by iamtarun1993**

### CODE REVIEW

**Comment by iamtarun1993** [comment]
iPhone x and iPhone 5 with 2 policy violations
Screenshot 2021-06-13 at 6 32 16 PM
Screenshot 2021-06-13 at 6 31 47 PM

**Comment by iamtarun1993** [comment]
iPhone 5 and iPhone X with more than 5- 10 violations.
Screenshot 2021-06-13 at 6 33 10 PM
Screenshot 2021-06-13 at 6 32 55 PM

**Review by Julias0 — APPROVED** [review]

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+$small-screen: 700px;
+
+.critical-policy-violation {
+  padding: 10px 15px 0 15px;
+
+  &--header {
+    color: $black;
+    font-weight: 500;
+    font-size: 18px;
+    text-align: center;
```
Comment (abhishek1234321) [inline src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:11]: can we use a `utility` class like we do in webapp? `text-center`?

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    line-height: 23px;
+    padding: 14px 0;
+  }
+
+  &--info {
+    color: $blue-black;
+    font-weight: 400;
```
Comment (abhishek1234321) [inline src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:18]: isn't this `400` by default?

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    font-size: 14px;
+    text-align: center;
```
Comment (abhishek1234321) [inline src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:20]: this is repeated, could be reused

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    line-height: 18px;
+    padding-bottom: 18px;
+  }
+
+  &--content {
+    --height: calc(100% - 288px);
```
Comment (abhishek1234321) [inline src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:26]: why `288`?

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  &--body {
+    background: $white-1;
+    border: 1px solid $grey-1;
+    box-sizing: border-box;
+    border-radius: 8px;
+    max-height: 40vh;
```
Comment (abhishek1234321) [inline src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:34]: why 40?

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      @media (max-height: $small-screen) {
+        max-height: 27vh;
+      }
+    overflow-y: auto;
+    padding: 2px 13px 2px 5px;
```
Comment (abhishek1234321) [inline src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:39]: this padding value is quite odd and asymmetrical

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  &--error-icon {
+    height: 20px;
+    width: 20px;
+  }
+
+  &--list {
+    margin: 0;
```
Comment (abhishek1234321) [inline src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:48]: can we not use `m-0` and `p-0` utility classes? if they aren't already present, please add them

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  }
+
+  &--cta {
+    margin-bottom: 5px;
+  }
+
+  &--secondary-cta {
+    height: 100%;
+    width: 90%;
```
Comment (abhishek1234321) [inline src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:66]: what is this about?

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    background: transparent;
+    color: $black;
+    text-align: center;
+    font-size: 14px;
```
Comment (abhishek1234321) [inline src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:70]: repeated

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense.module.ts [hunk 10]
```diff
@@ -18,7 +18,6 @@ import { MatDatepickerModule } from '@angular/material/datepicker';
 import { SharedModule } from 'src/app/shared/shared.module';
```
[the remark this answers is not in the document]
Comment (iamtarun1993) [reply src/app/fyle/add-edit-expense/add-edit-expense.module.ts:18]: After clicking on the "Continue" we are closing the popup instantly . We are not doing any API call here.
So we shouldn't show any loader on "Continue".

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss (hunk 4 above)
Comment (iamtarun1993) [reply src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:23]: most probably this is not needed now.

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss (hunk 5 above)
Comment (iamtarun1993) [reply src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:31]: The list of expense violations screen should be 40% visible at max for the larger screen.
Else it should be scrollable.

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss (hunk 8 above)
Comment (iamtarun1993) [reply src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:61]: We are not using the ionic button here, by default button will be sized based on the text, here we want to increase the button size to 90% of the grid.
Screenshot 2021-06-14 at 3 53 36 PM

Screenshot 2021-06-14 at 3 54 36 PM

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss (hunk 6 above)
Comment (iamtarun1993) [reply src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:36]: Screenshot 2021-06-14 at 4 00 41 PM

File: src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss (hunk 9 above)
Comment (iamtarun1993) [reply src/app/shared/components/fy-critical-policy-violation/fy-critical-policy-violation.component.scss:64]: was it for font-size or text-center?

**Review by abhishek1234321 — APPROVED** [review]

**Pull request merged**